src/data_source/storage/sql.rs

```diff
@@ -1789,6 +1800,534 @@ impl SqlStorage {
+                        Box::new(*block as i64),
+                    ],
+                )
+                .await
+            }
+        }?
+        .map(|result| match result {
+            Ok(row) => parse_block::<Types>(row),
+            Err(err) => Err(err),
+        });
+
+        let transaction_summary_stream = block_stream.flat_map(|row| match row {
+            Ok(block) => stream::iter(
+                block
+                    .enumerate()
+                    .enumerate()
+                    .map(|(index, (_, txn))| {
+                        TransactionSummary::try_from((&block, index, txn)).map_err(|err| {
+                            QueryError::Error {
+                                message: err.to_string(),
+                            }
+                        })
+                    })
+                    .collect::<Vec<QueryResult<TransactionSummary<Types>>>>()
+                    .into_iter()
+                    .rev()
+                    .collect::<Vec<QueryResult<TransactionSummary<Types>>>>(),
+            ),
+            Err(err) => stream::iter(vec![Err(err)]),
+        });
+
+        let transaction_summary_vec = transaction_summary_stream
+            .try_collect::<Vec<TransactionSummary<Types>>>()
+            .await?;
+
+        Ok(transaction_summary_vec
+            .into_iter()
+            .skip_while(|txn| {
+                if let TransactionIdentifier::Hash(hash) = target {
+                    txn.hash != *hash
+                } else if let TransactionIdentifier::HeightAndOffset(height, offset) = target {
+                    txn.height != *height as u64 && txn.offset != *offset as u64
+                } else {
+                    false
+                }
+            })
+            .collect::<Vec<TransactionSummary<Types>>>())
+    }
+
+    async fn get_transaction_detail(
+        &self,
+        request: &TransactionIdentifier<Types>,
+    ) -> QueryResult<TransactionDetailResponse<Types>> {
+        let target = request;
+
+        let (query, params): (String, Vec<Box<dyn ToSql + Send + Sync>>) = match target {
+            TransactionIdentifier::Latest => (
+                format!(
+                    "SELECT {BLOCK_COLUMNS}
+                            FROM header AS h
+                            JOIN payload AS p ON h.height = p.height
+                            WHERE h.height = (
+                                SELECT MAX(t1.block_height)
+                                    FROM transaction AS t1
+                            )
+                            ORDER BY h.height DESC"
+                ),
+                vec![],
+            ),
+            TransactionIdentifier::HeightAndOffset(height, offset) => (
+                format!(
+                    "SELECT {BLOCK_COLUMNS}
+                            FROM header AS h
+                            JOIN payload AS p ON h.height = p.height
+                            WHERE h.height = (
```

Comment:
the subquery can return multiple rows so this should use **IN** clause?

Reply:
There's a `LIMIT 1` on this subquery, so this subquery can only return a single row.